crates/secrets/src/aws_secrets_manager.rs

```diff
@@ -32,12 +56,99 @@ impl AwsSecretsManager {
     pub fn new() -> Self {
         Self {}
     }
+
+    /// Initializes AWS configuration and verifies AWS credentials.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error:
+    /// - If the AWS configuration cannot be loaded.
+    /// - If the call to STS `get_caller_identity` fails, which might be due to invalid or expired AWS credentials.
+    pub async fn init(self) -> Result<()> {
+        let config = aws_config::defaults(BehaviorVersion::v2023_11_09())
+            .load()
+            .await;
+
+        let sts_client = aws_sdk_sts::Client::new(&config);
+
+        if sts_client.get_caller_identity().send().await.is_err() {
+            return UnableToVerifyAwsIdentitySnafu {}.fail();
+        }
```

Comment:
This should use `.context()` - otherwise we are losing the inner error which might explain why the creds are invalid.

Reply:
Updated 👍